gradle-conjure-api/src/main/java/com/palantir/gradle/conjure/api/EndpointVersionBound.java

```diff
@@ -58,12 +60,9 @@ public void setMinVersion(String minVersion) {
         this.minVersion = minVersion;
     }
 
+    @JsonInclude(Include.NON_NULL)
```

Comment:
Can we add round-trip serialization of this class with and without maxVersion set, for both JSON and java serialization?

Does the `JsonInclude` annotation belong with the `JsonProperty` annotation above on line 36?

Reply:
Sorry, had to step away for a bit after pushing the PR. Will add some tests.

Reply:
Turns out I did add a test (at least one direction of the round-trip, and may need some updates): https://github.com/palantir/gradle-conjure/blob/a387dcb6d0169833c5c15f1d5403e5d3bac2c156/gradle-conjure/src/test/groovy/com/palantir/gradle/conjure/ConjureServiceDependencyTest.groovy#L383
 
But are the tests actually running? I'm only seeing java test results: https://app.circleci.com/pipelines/github/palantir/gradle-conjure/1608/workflows/797c8a2a-591d-4e8e-a4e7-338ac8df6a07/jobs/11399/artifacts

Reply:
I didn't see any other tests being run locally either but I was able to run all the tests with some gradle hackery I googled. I haven't added the gradle fix because I didn't spend too much time to figure out if it's a correct fix, but with the following change

```
diff --git a/gradle-conjure/build.gradle b/gradle-conjure/build.gradle
index b091199..4f48f77 100644
--- a/gradle-conjure/build.gradle
+++ b/gradle-conjure/build.gradle
@@ -33,6 +33,8 @@ dependencies {
     annotationProcessor 'org.immutables:value'
     compileOnly 'org.immutables:value::annotations'
 
+    testImplementation "org.junit.jupiter:junit-jupiter:5.7.0"
+    testRuntimeOnly "org.junit.vintage:junit-vintage-engine:5.7.0"
     testImplementation gradleTestKit()
     testImplementation 'com.netflix.nebula:nebula-test'
     testImplementation 'com.squareup.okhttp3:mockwebserver'
@@ -57,3 +59,12 @@ tasks.withType(JavaCompile).configureEach {
         check('PublicConstructorForAbstractClass', CheckSeverity.OFF)
     }
 }
+
+test {
+    description = "Runs JUnit 4 tests."
+    useJUnit()
+    scanForTestClasses = false
+    include(["**/*Tests.class", "**/*Test.class"])
+    exclude(["**/testng/**/*.*", "**/jupiter/**/*.*"])
+    reports.junitXml.destination = file("$buildDir/test-results")
+}
```

there are 2 unrelated tests failing:
ConjurePluginTest. works with checkUnusedDependencies
ConjurePublishTypeScriptTest. publishes generated code

Reply:
^ Resolved in PR #885